compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirSupertypesResolution.kt

```diff
@@ -129,8 +132,22 @@ private class FirApplySupertypesTransformer(
             "Expected single supertypeRefs, but found ${supertypeRefs.size} in ${typeAlias.symbol.classId}"
         }
 
+        val supertypeRef = supertypeRefs.single()
+        // TODO: Need better handling for type alias with type parameters.
+        if (typeAlias.typeParameters.isEmpty()) {
+            (supertypeRef.firClassLike(session) as? FirRegularClass)?.let {
```

Comment:
NB: It can be not a regular class, but for example a different type alias

Reply:
Thanks. Also noticed that it could be type parameter. Changed to use `fullyExpandedType`, and deferred to symbolize it.